### ISSUE TITLE
Convertions can't compare BigInt to numerics

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#654]
```
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:213: error: type mismatch:
[info]  found   : Byte(-128)
[info]  required: scala.math.BigInt
[info]     val isByte = bi >= Byte.MinValue && bi <= Byte.MaxValue
[info]                             ^
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:213: error: type mismatch:
[info]  found   : Byte(127)
[info]  required: scala.math.BigInt
[info]     val isByte = bi >= Byte.MinValue && bi <= Byte.MaxValue
[info]                                                    ^
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:214: error: type mismatch:
[info]  found   : Short(-32768)
[info]  required: scala.math.BigInt
[info]     val isShort = bi >= Short.MinValue && bi <= Short.MaxValue
[info]                               ^
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:214: error: type mismatch:
[info]  found   : Short(32767)
[info]  required: scala.math.BigInt
[info]     val isShort = bi >= Short.MinValue && bi <= Short.MaxValue
[info]                                                       ^
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:215: error: type mismatch:
[info]  found   : Char('\00')
[info]  required: scala.math.BigInt
[info]     val isChar = bi >= Char.MinValue && bi <= Char.MaxValue
[info]                             ^
[info] /Users/dark/workspace/dotty/tests/partest-generated/run/is-valid-num.scala:215: error: type mismatch:
[info]  found   : Char('￿')
[info]  required: scala.math.BigInt
[info]     val isChar = bi >= Char.MinValue && bi <= Char.MaxValue
```

**Comment by odersky** [comment]
This is a consequence of the fact that we dropped weak conformance in Dotty. BigInt has an implicit conversion from Int to BigInt, but none from Byte, Short, Char. And we cannot chain implicit conversions.

The proper solution should be to give BigInt the missing conversions.
I am adding them to DottyPredef for the moment.

**Comment by smarter** [comment]
Fixed by #902, leaving open until we move this test out of pending.
